### PR TITLE
feat: export private key

### LINKED ISSUE #543: Allow user to retrieve private key for a specific key pair
## Story

**As a user**
**I want to be able to retrieve the private key for a specific keypair within my wallet**
**So that I can use individual keys in other wallet systems or applications without sharing my full seed phrase, and use keys from my wallet in systems that don't support BIP39 key generation.**

## Acceptance Criteria

- [ ] I can select a wallet and key and choose an option to reveal the private key 
- [ ] I must enter my password before being shown the key
- [ ] I can click to copy the private key
- [ ] I am warned not to share this key with anyone etc.

## Tasks

- [ ] Detailed sketch
- [ ] Visual design
- [ ] Frontend tasks:
    - [ ] Implement UI
    - [ ] Integrate with backend
    - [ ] Write unit tests
- [ ] Backend tasks:
    - [ ] TBC
    - [ ] Write unit tests
- [ ] Write end to end tests
- [ ] Code review
- [ ] Team verify acceptance criteria met
- [ ] Exploratory QA testing 
- [ ] Capture bugs and new edge cases in tickets (this one or new TBD) 
- [ ] Write automated regression tests for the A/Cs (expanding on A/Cs where possible) 
- [ ] Add story and A/Cs to spec doc 
- [ ] Review with stakeholders
- [ ] Release

## Notes
Screenshot 2023-09-24 at 16 30 50
Screenshot 2023-09-24 at 16 30 56
Screenshot 2023-09-24 at 16 31 02
Screenshot 2023-09-24 at 16 31 41

### PR DESCRIPTION
**Pull request by dexturr**
Closes #543 

https://github.com/vegaprotocol/vegawallet-browser/assets/26136207/9ea559e3-38d4-46f8-936a-7176615fb8b1

### CODE REVIEW

**Comment by dalebennett1992** [comment]
Initial feedback/ potential issues:

### Display of export private key
image

The display of export private key looks a little off and I think it is due to a few things:
1. Full stop.  This should not have one IMO
2. The arrow. Should it be underlined? It also looks quite a lot bigger than the text, also should it be spaced a little? I am not sure exactly what makes it look odd but it feels like it might be one or a combination of those things.

Sanity question: Do we want it right at the bottom? Or do we want this option closer to the other key options? (copy, block explorer), I get why you might want to single this action out. But feels worth asking.

### Display of the export/close buttons
image


Export and Close are not aligned with eachother. Stands out to my eyes.

### Export spins out
Been going for about 2 mins so far.
image

shout if you wanna go through this, assuming you will for the last point in particular.

**Review by dalebennett1992 — CHANGES_REQUESTED** [review]
as per comment

File: specs/1129-HDCN-hidden_container.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## Hidden information container
```
Comment (dalebennett1992) [inline specs/1129-HDCN-hidden_container.md:1]: nicely done

File: specs/1128-EXPT-export_private_key.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## Export private key
```
Comment (dalebennett1992) [inline specs/1128-EXPT-export_private_key.md:1]: if we are being picky there is probably a missing (but simple) key details AC. Something like:

'There is a way to export a private key'

Then just whack the link into the key details unit test checking the element exists

File: specs/1128-EXPT-export_private_key.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+- I can see a notifications warning me about exporting my private keys (<a name="1128-EXPT-001" href="#1128-EXPT-001">1128-EXPT-001</a>)
+- There is a close button on the modal(<a name="1128-EXPT-002" href="#1128-EXPT-002">1128-EXPT-002</a>)
+- I am required to enter my passphrase to export my private keys (<a name="1128-EXPT-003" href="#1128-EXPT-003">1128-EXPT-003</a>)
+- If my passphrase is incorrect I see an error informing me of this (<a name="1128-EXPT-004" href="#1128-EXPT-004">1128-EXPT-004</a>)
+- If I enter the correct passphrase then I see the [reveal hidden information component](./1129-HDCN-hidden_container.md) and a button to close the modal (<a name="1128-EXPT-005" href="#1128-EXPT-005">1128-EXPT-005</a>)
```
Comment (dalebennett1992) [inline specs/1128-EXPT-export_private_key.md:7]: additional: If my password is blank the export button will remain disabled (and corresponding unit test please)

**Comment by dexturr** [comment]
@dalebennett1992 All comments resolved

**Comment by dalebennett1992** [comment]
More feedback after @emilbayes changes
image

So this seems like a mash of errors, I think we only need 'Error: Invalid passphrase'
